File: android/src/main/java/expo/modules/libsignal/PreKeyRecords.kt
package expo.modules.libsignal

import expo.modules.kotlin.sharedobjects.SharedObject
import org.signal.libsignal.protocol.state.KyberPreKeyRecord
import org.signal.libsignal.protocol.state.PreKeyRecord
import org.signal.libsignal.protocol.state.SignedPreKeyRecord

class PreKeyRecordRef(val record: PreKeyRecord) : SharedObject()

class SignedPreKeyRecordRef(val record: SignedPreKeyRecord) : SharedObject()

class KyberPreKeyRecordRef(val record: KyberPreKeyRecord) : SharedObject()

File: ios/PreKeyRecords.swift
import Foundation
import ExpoModulesCore
import LibSignalClient

final class PreKeyRecordRef: SharedObject {
  let record: PreKeyRecord

  init(record: PreKeyRecord) {
    self.record = record
    super.init()
  }
}

final class SignedPreKeyRecordRef: SharedObject {
  let record: SignedPreKeyRecord

  init(record: SignedPreKeyRecord) {
    self.record = record
    super.init()
  }
}

final class KyberPreKeyRecordRef: SharedObject {
  let record: KyberPreKeyRecord

  init(record: KyberPreKeyRecord) {
    self.record = record
    super.init()
  }
}
